samples/csharp/getting-started/DatabaseIntegration/DatabaseIntegration/Program.cs

```diff
@@ -0,0 +1,121 @@
+namespace DatabaseIntegration
+{
+    public class Program
+    {
+        // The url for the dataset that will be downloaded
+        public static string datasetUrl = "https://raw.githubusercontent.com/dotnet/machinelearning/244a8c2ac832657af282aa312d568211698790aa/test/data/adult.train";
+
+        public static IEnumerable<string> ReadRemoteDataset(string url)
+        {
+            using(var client = new WebClient())
+            using(var stream = client.OpenRead(url))
+            using(var reader = new StreamReader(stream))
+            {
+                string line;
+                while ((line = reader.ReadLine()) != null)
+                {
+                    yield return line;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Populates the database with the specified dataset url.
+        /// </summary>
+        public static void CreateDatabase(string url)
+        {
+            var dataset = ReadRemoteDataset(url);
+            using (var db = new AdultCensusContext())
+            {
+                // Ensure that we have a clean database to start with.
+                db.Database.EnsureDeleted();
+                db.Database.EnsureCreated();
+                Console.WriteLine($"Database created, populating...");
+
+                // Parse the dataset.
+                var data = dataset
+                    .Skip(1) // Skip the header row
+                    .Select(l => l.Split(','))
+                    .Where(row => row.Length > 1)
+                    .Select(row => new AdultCensus()
+                    {
+                        Age = int.Parse(row[0]),
+                        Workclass = row[1],
+                        Education = row[3],
+                        MaritalStatus = row[5],
+                        Occupation = row[6],
+                        Relationship = row[7],
+                        Race = row[8],
+                        Sex = row[9],
+                        CapitalGain = row[10],
+                        CapitalLoss = row[11],
+                        HoursPerWeek = int.Parse(row[12]),
+                        NativeCountry = row[13],
+                        Label = (int.Parse(row[14]) == 1) ? true : false
+                    });
+
+                // Add the data into the database
+                db.AdultCensus.AddRange(data);
+
+                var count = db.SaveChanges();
+                Console.WriteLine($"Total count of items saved to database: {count}");
+            }
+        }
+
+        public static void Main()
+        {
+            // Seed the database with the dataset.
+            CreateDatabase(datasetUrl);
+
+            using(var db = new AdultCensusContext())
+            {
+                // Set the query tracking behavior to be NoTracking, this is particularly useful because
+                // our scenarios only require read-only access.
+                db.ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTracking;
```

Comment:
I have a slight preference for having AsNotracking() call in-line in the query instead. My reasoning:

- It is a simpler API 
- Presumably some customers will want to use some data in their database to train an ML model, while keeping the rest of their EF Core usage untouched
- If you are copying and pasting the key code from this sample into your own app, the chances that you will copy AsNoTracking() seem a bit higher